roles/elasticsearch/tasks/main.yml

```diff
@@ -39,4 +39,9 @@
 - name: register elasticsearch firewall ports
   set_fact:
     firewall_ports: >
-      {{ firewall_ports + [elasticsearch_port] }}
+      {{ firewall_ports }} + [
+        "{{ item }}:{{ elasticsearch_port }}"
+      ]
+  with_items:
+    - "{{ groups['kibana_hosts']|host_addresses(hostvars, 'ipv4') }}"
```

Comment:
I'm having second thoughts about including this sort of logic in the roles themselves.  I am thinking that the roles should be usable in isolation, and logic like this should go into `group_vars` somewhere (e.g., `group_vars/elastic_hosts.yml` or something).

Separately, there should be a configuration parameter for the role that is "the ip address to which clients should connect", and this is the address that should be used in the firewall -- not a list of default addresses from the hosts.